core/src/utils/animation/animation.ts

```diff
@@ -950,21 +840,12 @@ export const createAnimation = (animationId?: string): Animation => {
          * element too quickly
          */
         raf(() => {
-          clearCSSAnimationPlayState();
```

Comment:
Do we still need the code above? It looks to be specific to CSS animations and display: none

Reply:
Some level of fallback behavior is required to satisfy existing unit tests: https://github.com/ionic-team/ionic-framework/actions/runs/8333321414/job/22804541495.

I think we have two options:
1. Have the minimal code to satisfy the unit test: https://github.com/ionic-team/ionic-framework/pull/29123/commits/72a96622c99c2d5271ba277fa784e31eb66962e3
2. Remove the unit tests

I've done option 1 to reduce the number of changes. However I question the value of these tests at all. They would have only tested the fallback CSS behavior, not the web animation behavior. 

If you are in favor, I'd propose removing those (can be either part of this PR or a separate ticket).